### ISSUE TITLE
Exception stack lost when unexpected error happened in processing MQTT Message

### ISSUE THREAD

**Issue by fjcyue** [moquette-io/moquette#380]
Recently，we found log like this in our server：

> 2017-11-23 16:25:46.324 [nioEventLoopGroup-3-2] ERROR i.m.server.netty.NettyMQTTHandler - An unexpected exception was caught while processing MQTT message. Closing Netty channel. CId=00000000001, cause=null, errorMessage=null

the real cause is lost，so we try to review the `NettyMQTTHandler ` to find out what happened。

```
   @Override
    public void channelRead(ChannelHandlerContext ctx, Object message) {
        try {
              // omited .......
        } catch (Throwable ex) {
            LOG.error("Exception was caught while processing MQTT message, " + ex.getCause(), ex);
            ctx.fireExceptionCaught(ex);
            ctx.close();
        } finally {
            ReferenceCountUtil.release(msg);
        }
    }

    @Override
    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
        LOG.error("An unexpected exception was caught while processing MQTT message. Closing Netty channel. CId={}, " +
            "cause={}, errorMessage={}", NettyUtils.clientID(ctx.channel()), cause.getCause(), cause.getMessage());
        ctx.close();
    }
```

the `exceptionCaught()` method only print the `cause ` and `message ` of the original  throwable, so the stack is lost.

we search the history of this file, and found the version of exceptionCaught() before commit 1f1bfff  seems more graceful, is that so?

and，the  `ctx.close()` in the catch block of `channelRead()` seems duplicate since `fireExceptionCaught()` invoked and  `exceptionCaught()` also close it. this is added in commit fcb7870 by @AndreasMager  and the comment doesn't show the reason. I'm not sure if it can be removed.

**Comment by davidgraeff** [comment]
Any news on this? I have the same messages in my log and have no idea what went wrong.

**Comment by andsel** [comment]
I plan to fix in the next days and create the rel 0.11

**Comment by andsel** [comment]
Fixed with 3c9cc5fa8ee8dd340f74cd684ed29e839f5f5609
